build.cake

```diff
@@ -0,0 +1,198 @@
+#addin "Cake.XdtTransform"
+#addin "Cake.Powershell"
+#addin "Cake.Http"
+#addin "Cake.Json"
+#addin "Newtonsoft.Json"
+
+#load "local:?path=CakeScripts/helper-methods.cake"
+
+
+var target = Argument<string>("Target", "Default");
+var configuration = new Configuration();
+var cakeConsole = new CakeConsole();
+
+var unicornSyncScript = $"./scripts/Unicorn/Sync.ps1";
+
+var configJsonFile = "cake-config.json";
+
+/*===============================================
+================ MAIN TASKS =====================
+===============================================*/
+
+Setup(context =>
+{
+	cakeConsole.ForegroundColor = ConsoleColor.Yellow;
+	
+    var configFile = new FilePath(configJsonFile);
+    configuration = DeserializeJsonFromFile<Configuration>(configFile);
+});
+
+Task("Default")
+.WithCriteria(configuration != null)
+.IsDependentOn("Clean")
+.IsDependentOn("Copy-Sitecore-Lib")
+.IsDependentOn("Publish-Website-Projects")
+.IsDependentOn("Apply-Xml-Transform")
+.IsDependentOn("Modify-Unicorn-Source-Folder")
+.IsDependentOn("Publish-Transforms")
+.IsDependentOn("Sync-Unicorn")
+.IsDependentOn("Deploy-EXM-Campaigns");
+
+Task("Quick-Deploy")
+.WithCriteria(configuration != null)
+.IsDependentOn("Clean")
+.IsDependentOn("Copy-Sitecore-Lib")
+.IsDependentOn("Publish-Website-Projects")
+.IsDependentOn("Apply-Xml-Transform")
+.IsDependentOn("Modify-Unicorn-Source-Folder")
+.IsDependentOn("Publish-Transforms");
+
+Task("Initial")
+.WithCriteria(configuration != null)
+.IsDependentOn("Clean")
+.IsDependentOn("Copy-Sitecore-Lib")
+.IsDependentOn("Publish-Website-Projects")
+.IsDependentOn("Apply-Xml-Transform")
+.IsDependentOn("Modify-Unicorn-Source-Folder")
+.IsDependentOn("Publish-Transforms")
+.IsDependentOn("Initial-From-Sync-Unicorn");
+
+
+Task("Publish")
+.WithCriteria(configuration != null)
+.IsDependentOn("Clean")
+.IsDependentOn("Copy-Sitecore-Lib")
+.IsDependentOn("Publish-Website-Projects")
+.IsDependentOn("Apply-Xml-Transform")
+.IsDependentOn("Modify-Unicorn-Source-Folder")
+.IsDependentOn("Publish-Transforms");
+
+
+/*===============================================
+================= SUB TASKS =====================
+===============================================*/
+
+
+
+Task("Clean").Does(() => {
+    CleanDirectories($"{configuration.SourceFolder}/**/obj");
+    CleanDirectories($"{configuration.SourceFolder}/**/bin");
+});
+
+Task("Copy-Sitecore-Lib").Does(() => {
```

Comment:
Need a task to validate the ./lib/Modules/Commerce exists _or_ need to add placeholder readme.txt file in an empty folder.
If folder doesn't exist the Copy-Sitecore-Lib task fails

Reply:
Should we create this "./lib/Modules/Commerce" automatically?

Reply:
(I don't know why my comments ended up in the wrong thread)
Yes I would prefer to create the folder if it doesn't exist.